src/Mapper/AutoConverterProcessor.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bindto\Mapper;
+
+use Bindto\Annotation\AutoConvert;
+use Bindto\Annotation\ConvertAnnotationInterface;
+use Bindto\ConverterInterface;
+use function Functional\first;
+use Doctrine\Common\Collections\Collection;
+use phpDocumentor\Reflection\DocBlock\Tags\Var_;
+use phpDocumentor\Reflection\DocBlockFactory;
+use phpDocumentor\Reflection\DocBlockFactoryInterface;
+use phpDocumentor\Reflection\Types\Array_;
+use phpDocumentor\Reflection\Types\Compound;
+use phpDocumentor\Reflection\Types\ContextFactory;
+use phpDocumentor\Reflection\Types\Null_;
+use ReflectionProperty;
+use Ursula\Common\Exception\DomainException;
+
+/**
+ * Mapper that processes @AutoConvert annotations for a property.
+ */
+class AutoConverterProcessor
+{
+    /**
+     * @var DocBlockFactoryInterface
+     */
+    private $docBlockFactory;
+
+    /**
+     * @var ContextFactory
+     */
+    private $docBlockContextFactory;
+
+    /**
+     * @var ConvertingObjectMapper
+     */
+    private $convertingObjectMapper;
+
+    public function __construct(ConvertingObjectMapper $convertingObjectMapper)
+    {
+        $this->convertingObjectMapper = $convertingObjectMapper;
+        $this->docBlockFactory = DocBlockFactory::createInstance();
+        $this->docBlockContextFactory = new ContextFactory();
+    }
+
+    public function process(ConvertAnnotationInterface $annotation, ReflectionProperty $property): array
+    {
+        $typeInfo = $this->extractTypeInformation($property);
+        $converters = $this->convertingObjectMapper->getConverters();
+
+        $converter = first($converters, function (ConverterInterface $converter) use ($typeInfo) {
+            return $converter->canProduceType($typeInfo->typeName);
+        });
+
+        return $converter->autoconfigure($annotation, $typeInfo->typeName, $typeInfo->isArray, $typeInfo->isNullable);
+    }
+
+    private function extractTypeInformation(ReflectionProperty $property): TypeInfo
+    {
+        $typeInfo = new TypeInfo();
+
+        $docBlock = $this->docBlockFactory->create(
+            $property->getDocComment(),
+            $this->docBlockContextFactory->createFromReflector($property)
+        );
+
+        if (! $docBlock->hasTag('var')) {
+            throw new DomainException(
+                sprintf('Property "%s" on "%s" must define a type', $property->getName(), $property->getDeclaringClass())
+            );
+        }
+
+        /* @var Var_ $var */
+        $var = $docBlock->getTagsByName('var')[0];
```

Comment:
I see you've got the php functional library in this project. Rather use `first`.

Reply:
Done